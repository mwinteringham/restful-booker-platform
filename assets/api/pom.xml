<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>com.automationintesting</groupId>
    <artifactId>restful-booker-platform-assets</artifactId>
    <version>1.6.${revision}</version>

    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.7.5</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <revision>SNAPSHOT</revision>
        <skipVisual>true</skipVisual>
    </properties>

    <profiles>
        <profile>
            <id>ci</id>
            <activation>
                <property>
                    <name>env.skipVisual</name>
                </property>
            </activation>
            <properties>
                <skipVisual>false</skipVisual>
            </properties>
        </profile>
    </profiles>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <release>15</release>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <configuration>
                    <classifier>exec</classifier>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>repackage</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.22.0</version>
                <configuration>
                    <includes>
                        <include>**/ServeContentTest.java</include>
                    </includes>
                </configuration>
            </plugin>
            <plugin>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>2.22.0</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <includes>
                        <include>**/AuthVisualTest.java</include>
                        <include>**/RoomVisualTest.java</include>
                        <include>**/HomeVisualTest.java</include>
                    </includes>
                    <skipITs>${skipVisual}</skipITs>
                </configuration>
            </plugin>
        </plugins>
    </build>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-actuator</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter</artifactId>
            <version>5.9.1</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured</artifactId>
            <version>5.3.0</version>
            <exclusions>
                <!-- Exclude Groovy because of dependency issue -->
                <exclusion>
                    <groupId>org.codehaus.groovy</groupId>
                    <artifactId>groovy-xml</artifactId>
                </exclusion>
            </exclusions>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.approvaltests</groupId>
            <artifactId>approvaltests</artifactId>
            <version>18.5.0</version>
        </dependency>
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>4.6.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.applitools</groupId>
            <artifactId>eyes-selenium-java3</artifactId>
            <version>5.0.0</version>
        </dependency>
        <dependency>
            <groupId>com.xebialabs.restito</groupId>
            <artifactId>restito</artifactId>
            <version>1.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.14.0</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.14.0</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-annotations</artifactId>
            <version>2.14.0</version>
        </dependency>
        <dependency>
            <groupId>io.github.bonigarcia</groupId>
            <artifactId>webdrivermanager</artifactId>
            <version>5.3.1</version>
        </dependency>
    </dependencies>
    
</project>