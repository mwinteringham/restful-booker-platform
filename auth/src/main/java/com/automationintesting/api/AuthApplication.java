package com.automationintesting.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan(basePackages = "com.automationintesting")
public class AuthApplication {

    public static void main(String[] args) {
        SpringApplication.run(AuthApplication.class, args);
    }

}
