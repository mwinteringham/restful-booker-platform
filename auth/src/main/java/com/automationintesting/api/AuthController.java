package com.automationintesting.api;

import com.automationintesting.model.Auth;
import com.automationintesting.model.Decision;
import com.automationintesting.model.Token;
import com.automationintesting.service.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletResponse;
import java.sql.SQLException;

@RestController
public class AuthController {

    @Autowired
    private AuthService authService;

    @RequestMapping(value = "/login", method = RequestMethod.POST)
    public ResponseEntity<Token> createToken(@RequestBody Auth auth, HttpServletResponse response) throws SQLException {
        Decision decision = authService.queryCredentials(auth);

        if(decision.getStatus() == HttpStatus.OK){
            Cookie cookie = new Cookie("token", decision.getToken().getToken());
            cookie.setPath("/");

            response.addCookie(cookie);

            return ResponseEntity.ok().build();
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(null);
        }
    }

    @RequestMapping(value = "/validate", method = RequestMethod.POST)
    public ResponseEntity<Token> validateToken(@RequestBody Token token) throws SQLException {
        HttpStatus httpStatus = authService.verify(token);

        return ResponseEntity.status(httpStatus).build();
    }

    @RequestMapping(value = "/logout", method = RequestMethod.POST)
    public ResponseEntity clearToken(@RequestBody Token token) throws SQLException {
        HttpStatus httpStatus = authService.deleteToken(token);

        return ResponseEntity.status(httpStatus).build();
    }

}
