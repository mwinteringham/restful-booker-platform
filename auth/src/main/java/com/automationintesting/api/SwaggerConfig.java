package com.automationintesting.api;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.servers.Server;
import org.springdoc.core.GroupedOpenApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SwaggerConfig {

    @Bean
    public OpenAPI openAPI() {
        return new OpenAPI()
                .addServersItem(new Server().url("/auth/"));
    }

    @Bean
    public GroupedOpenApi publicApi() {
        return GroupedOpenApi.builder()
                .group("auth-api")
                .pathsToMatch("/**")
                .build();
    }

}
