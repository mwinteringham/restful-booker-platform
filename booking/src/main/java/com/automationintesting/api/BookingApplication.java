package com.automationintesting.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

import java.util.TimeZone;

@SpringBootApplication
@ComponentScan(basePackages = "com.automationintesting")
public class BookingApplication {

    public static void main(String[] args) {
        TimeZone.setDefault(TimeZone.getTimeZone("Etc/UTC"));

        SpringApplication.run(BookingApplication.class, args);
    }

}
