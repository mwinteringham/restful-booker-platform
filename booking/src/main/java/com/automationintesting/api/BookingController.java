package com.automationintesting.api;

import com.automationintesting.model.db.Booking;
import com.automationintesting.model.db.BookingSummaries;
import com.automationintesting.model.db.Bookings;
import com.automationintesting.model.db.CreatedBooking;
import com.automationintesting.model.service.BookingResult;
import com.automationintesting.service.BookingService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.sql.SQLException;
import java.util.Optional;

@RestController
public class BookingController {

    @Autowired
    private BookingService bookingService;

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public ResponseEntity getBookings(@RequestParam("roomid") Optional<String> roomid, @CookieValue(value ="token", required = false) String token) throws SQLException {
        BookingResult bookingResult = bookingService.getBookings(roomid, token);

        return ResponseEntity.status(bookingResult.getStatus()).body(bookingResult.getBookings());
    }

    @RequestMapping(value = "/", method = RequestMethod.POST)
    public ResponseEntity createBooking(@Valid @RequestBody Booking booking) throws SQLException {
        BookingResult bookingResult = bookingService.createBooking(booking);

        return ResponseEntity.status(bookingResult.getStatus()).body(bookingResult.getCreatedBooking());
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.GET)
    public ResponseEntity getBooking(@PathVariable(value = "id") int bookingId, @CookieValue(value ="token", required = false) String token) throws SQLException {
        BookingResult bookingResult = bookingService.getIndividualBooking(bookingId, token);

        return ResponseEntity.status(bookingResult.getStatus()).body(bookingResult.getBooking());
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.DELETE)
    public ResponseEntity deleteBooking(@PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        HttpStatus deleteHttpStatus = bookingService.deleteBooking(id, token);

        return ResponseEntity.status(deleteHttpStatus).build();
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.PUT)
    public ResponseEntity<CreatedBooking> updateBooking(@Valid @RequestBody Booking booking, @PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        BookingResult updateResult = bookingService.updateBooking(id, booking, token);

        return ResponseEntity.status(updateResult.getStatus()).body(updateResult.getCreatedBooking());
    }

    @RequestMapping(value = "/summary", method = RequestMethod.GET)
    public ResponseEntity getSummaries(@RequestParam("roomid") String roomid) throws SQLException {
        BookingSummaries bookingSummaries = bookingService.getBookingSummaries(roomid);

        return ResponseEntity.status(HttpStatus.OK).body(bookingSummaries);
    }

}
