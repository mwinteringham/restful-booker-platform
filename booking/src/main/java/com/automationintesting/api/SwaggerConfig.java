package com.automationintesting.api;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.servers.Server;
import org.springdoc.core.GroupedOpenApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SwaggerConfig {

    @Bean
    public OpenAPI openAPI() {
        return new OpenAPI()
                .addServersItem(new Server().url("/booking/"));
    }

    @Bean
    public GroupedOpenApi publicApi() {
        return GroupedOpenApi.builder()
                .group("booking-api")
                .pathsToMatch("/**")
                .build();
    }

}
