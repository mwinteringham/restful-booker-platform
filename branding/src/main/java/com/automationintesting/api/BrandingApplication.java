package com.automationintesting.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan(basePackages = "com.automationintesting")
public class BrandingApplication {

    public static void main(String[] args) {
        SpringApplication.run(BrandingApplication.class, args);
    }

}
