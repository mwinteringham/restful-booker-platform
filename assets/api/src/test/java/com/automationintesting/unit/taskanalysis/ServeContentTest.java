package com.automationintesting.unit.taskanalysis;

import com.automationintesting.UiApplication;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import static io.restassured.RestAssured.given;

// We need to start the app up to test it. So we use the SpringRunner class and SpringBootTest to configure
// and run the app.
@ExtendWith(SpringExtension.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, classes = UiApplication.class)
public class ServeContentTest {

    // We add the @Test annotation so that when JUnit runs it knows which
    // methods to run as tests
    @Test
    // We give the check a clear name to ensure that it is descriptive in
    // what it is checking
    public void checkUiServesStaticContent(){
        // Using rest-assured, we send an HTTP request to the UI API and
        // store the response
        Response response = given()
                              .get("http://localhost:3003/");

        // The response body is expected to be in HTML format. So rather than use
        // complex asserts we use ApprovalTests to verify the body string against
        // a previously saved body in the *.approved.txt file.
        Approvals.verify(response.prettyPrint());
    }

}
