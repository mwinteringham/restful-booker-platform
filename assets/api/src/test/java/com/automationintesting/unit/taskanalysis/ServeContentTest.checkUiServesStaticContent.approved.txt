<html lang="en">
  <head>
    <meta charset="UTF-8"/>
    <meta name="google" content="notranslate"/>
    <meta http-equiv="Content-Language" content="en"/>
    <title>Restful-booker-platform demo</title>
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"/>
    <link href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous"/>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js" integrity="sha384-wHAiFfRlMFy6i5SRaxvfOCifBUQy1xHdJ/yoi7FRNXMRBu5WHdZYu1hA6ZOblgut" crossorigin="anonymous"/>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js" integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k" crossorigin="anonymous"/>
    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet"/>
    <script defer="defer" src="js/main.js"/>
  </head>
  <body>
    <div id="root"/>
  </body>
</html>
